src/System.Memory/src/System/SpanHelpers.T.cs

```diff
@@ -120,8 +121,9 @@ public static unsafe int IndexOf<T>(ref T searchSpace, T value, int length)
         }
 
         public static bool SequenceEqual<T>(ref T first, ref T second, int length)
-            where T : struct, IEquatable<T>
+            where T : IEquatable<T>
         {
+            if (typeof(T) == typeof(byte)) return SequenceEqual(ref Unsafe.As<T, byte>(ref first), ref Unsafe.As<T, byte>(ref second), length);
```

Comment:
What makes this one special that it is special cased here?

Reply:
StartsWith calls this helper as well as SequenceEqual. Didn't want to duplicate the type check. I could change it if you think that would be cleaner.

Reply:
I think calling it here is bad for perf (for small sequences in particular).

Reply:
Good point. Will update.